tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php

```diff
@@ -157,6 +159,39 @@ public static function cleanDatabaseHardPrepareFeature()
         require_once _PS_ROOT_DIR_ . '/config/config.inc.php';
     }
 
+    /**
+     * This hook can be used to flag a feature for database hard reset
+     *
+     * @BeforeFeature @reset-context-before-feature
+     */
+    public static function resetContextBeforeFeature()
```

Comment:
There is already a mechanism for this which is automatic in `ContextFeatureContext::resetContext()` It relies on the `ContextMocker` class I think it would be more useful to improve this class with what you need This will benefice the whole tests rather than just this specific use case

Reply:
This improvement is still valid, it would improve the `ContextMocker` for all existing and future tests